remote/src/main/java/com/spotify/metrics/remote/OkRemote.java

```diff
@@ -49,14 +52,19 @@
     private final String host;
     private final int port;
     private final ObjectMapper mapper = new ObjectMapper();
+    private boolean closed = false;
 
     public OkRemote(String host, int port) {
         this.host = host;
         this.port = port;
     }
 
     @Override
-    public ListenableFuture<Integer> post(String path, String shardKey, Map jsonObj) {
+    public synchronized ListenableFuture<Integer> post(String path, String shardKey, Map jsonObj) {
+        if (closed) {
+            throw new RuntimeException("Calling post after shutdown");
```

Comment:
Is it possible that the user would have a separate thread providing metrics, and after the shutdown is called, this thread would continue and cause this exception to happen. Would the user expect instead to just have the metrics sending where ever just start yielding no-ops instead of causing terrible death? I would as a user prefer getting a warning message, and then just turn the post into no-op.